connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java

```diff
@@ -0,0 +1,323 @@
+            for (Map.Entry<String, KafkaFuture<Void>> entry : newResults.entrySet()) {
+                String topic = entry.getKey();
+                try {
+                    entry.getValue().get();
+                    log.info("Created topic {} on brokers at {}", topicsByName.get(topic), this);
+                } catch (ExecutionException e) {
+                    Throwable cause = e.getCause();
+                    if (cause instanceof UnsupportedVersionException) {
+                        // Broker is too old
+                        log.info("Unable to use Kafka admin client to create topics using the brokers at {}", this);
+                        return null;
+                    }
+                    if (e.getCause() instanceof TopicExistsException) {
+                        // Must have been created by another client, so keep going
+                        log.debug("Found existing topic '{}' on the brokers at {}", topic, this);
+                        continue;
+                    }
+                    throw e;
+                }
+            }
+            // Get descriptions for the topics we just tried to create, including any that did in fact exist
+            Map<String, TopicDescription> newDescriptions = describeTopics(topicsByName.keySet());
+            // And combine with the descriptions for the already-existing topics
+            descriptions.putAll(newDescriptions);
+            return descriptions;
+        } catch (InterruptedException e) {
+            Thread.interrupted();
+            throw new ConnectException("Interrupted while attempting to create/find topic(s) '" + requestedTopicNames + "'", e);
+        } catch (ExecutionException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof UnsupportedVersionException) {
+                // Broker is too old
+                log.debug("Unable to use Kafka admin client to create topic descriptions using the brokers at {}", this);
+                return null;
+            }
+            if (cause instanceof TimeoutException) {
+                // Timed out waiting for the operation to complete
+                throw new ConnectException("Timed out while checking for or creating topic(s) '" + requestedTopicNames + "'." +
+                        " This could indicate a connectivity issue, unavailable topic partitions, or if" +
+                        " this is your first use of the topic it may have taken too long to create.", cause);
+            }
+            throw new ConnectException("Error while attempting to create/find topics '" + requestedTopicNames + "'", cause);
+        }
+    }
+
+    @Override
+    public void close() {
+        admin.close();
+    }
+
+    @Override
+    public String toString() {
+        Object servers = adminConfig.get(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG);
+        return servers != null ? servers.toString() : "<unknown>";
```

Comment:
This method is currently used only in log messages, so rather than have a `toString()` that is potentially difficult to understand, it'd be better to get rid of it altogether and deal with the log messages in a different way.